website/blog/2023-03-23-audit-helper.md

```diff
@@ -0,0 +1,253 @@
+{% endset %}
+
+
+{% set new_etl_relation_query %}
+select * from {{ ref('dim_product') }}
+{% endset %}
+
+
+{% set audit_query = audit_helper.compare_column_values(
+    a_query=old_etl_relation_query,
+    b_query=new_etl_relation_query,
+    primary_key="product_id",
+    column_to_compare="status"
+) %}
+
+
+{% set audit_results = run_query(audit_query) %}
+
+
+{% if execute %}
+    {% do audit_results.print_table() %}
+{% endif %}
+```
+  
+The arguments used by this macro are pretty much the same used by the `compare_queries` macro, but with the addition of the `column_to_compare` argument, being that responsible, as the name suggests, for declaring which column is specifically going to be tested.
+
+Although similar to the compare_queries macro, this `compare_columns_value` macro gives great importance to the `primary_key` argument, which is going to be a fundamental part of generating correctly the metrics attributed to the compared column, acting as an anchor to compare every generated row from the refactored model with its legacy counterpart.
+
+Also, we can see that there is a table printing option enabled as default, so the result of the query gets printed on the terminal when the macro is run. That can be useful to quickly check out the compatibility status of a column, without leaving the code editor while refactoring SQL through dbt.
+
+3. Replace column names and source references from the example with the respective information of the compared models.
+4. Optional: Disable the `print_table()` command, so the model can be materialized on your data warehouse.
+    To disable table printing on terminal and to enable model materialization in your target data warehouse, the following can be done: replace this entire section of code, which declares the SQL model as variable and makes it printable, with the macro execution pattern.
+
+    ```sql
+    - Replace the commented code below:
+    - {% set audit_results = run_query(audit_query) %}
+
+    - {% if execute %}
+    - {% do audit_results.print_table() %}
+    - {% endif %}
+
+    - With the following piece of code:
+    {{ run_query(audit_query) }}
+    ```
+
+5. To get the results, you can simply run the model as you would with a regular dbt model using the following command:
+
+    `dbt run -select <name of your audit model>`
+
+    But different from the `compare_queries` macro, if you have keep t he printing function enabled, you should expect a table to be printed in the command line when you run the model, as portrayed in Figure 8. Otherwise, it will be materialized on your data warehouse like this:
+
+     <Lightbox src="/img/blog/2023-03-23-audit-helper/image3.png" title="Figure 8 — Example of table printed in command line" />
+
+    The `compare_column_values` macro separates column auditing results in seven different labels:
+    - **Perfect match**: count of rows (and relative percentage) where the column values compared between both tables are equal and not null;
```

Comment:
i suggested code formatting for the previous macro column names -- bold is good too! I think they should be consistent either way!